test/http/serve-delay-connect.go

```diff
@@ -0,0 +1,119 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/florianl/go-nfqueue"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultPort     = "8080"
+	defaultDelay    = time.Duration(40 * time.Second)
+	defaultQueueNum = 100
+)
+
+func NewServeDelayConnectCommand() *cobra.Command {
```

Comment:
I ran:
```
$ sudo ./ingress-operator serve-delay-connect-test-server
2024/06/12 16:32:46 Callback is registered with delay 40s
2024/06/12 16:32:46 Serving on 8080
```

Then I ran:
```
$ curl localhost:8080
HTTP/1.1
```

It didn't delay. Why is that? Do I need to run iptables commands for this function to work?

Reply:
Yes. A rule needs to send new tcp connections on `8080` port to the queue number used by the command (`100` by default). Let me add the exact command as a comment.

Reply:
Hmm, I ran it again, this time running:
```
$ sudo iptables -I INPUT -p tcp --dport 8080 -m conntrack --ctstate NEW -j NFQUEUE --queue-num 100
$ sudo ./ingress-operator serve-delay-connect-test-server
[sudo] password for gspence: 
2024/06/20 19:14:54 Callback is registered with delay 40s
2024/06/20 19:14:54 Serving on 8080
2024/06/20 19:14:56 Callback is triggered
2024/06/20 19:14:56 Delaying SYN packet for 40s
```
I curled in another terminal:
```
$ curl localhost:8080
HTTP/1.1
```
But immediately got an answer back. Seems like progress since I get log output for the `serve-delay-connect-test-server` command, but why would curl return immediately and not delay?

I suppose it could be some configuration with my laptop, because I did test with the route created by your E2E test, and it delays successfully. So I believe it works, but I was just curious why it wouldn't work on my local laptop.

EDIT: I figured it out. I need to use `curl 127.0.0.1:8080`, then it works. I wiresharked it and looks like curl used ipv6, which didn't work correctly. Not sure if it's worth a comment somewhere that IPv6 doesn't work correctly.

Reply:
Right, it must be laptop specific because it works on my old RHEL. The golang's server is supposed to bind on ipv6 and ipv4 addresses for localhost, so the problem should be in the iptables. Would this work for you if the iptables command is run for ipv6?
```
sudo ip6tables -I INPUT -p tcp --dport 8080 -m conntrack --ctstate NEW -j NFQUEUE --queue-num 100
```

Reply:
Ah yea that worked. Whoops, I forgot that iptables is ipv4 ipv6 specific.